Capstone/Assets/Game/Scripts/AbstractClasses/Character.cs

```diff
@@ -92,6 +92,8 @@ public virtual bool TakeDamage(int damagingID, float damage)
         protected abstract void SwitchWeapon(int weaponSelect);
         protected abstract void Die();
 
+        public abstract void Respawn();
```

Comment:
Why is the Respawn method here? Are we planning on using it for the enemies too or just a Spawn method for them?